### ISSUE TITLE
Is it normal that betwennessCentrality measure with edge weights gives a higher score to the nodes with less edge weights? 

### ISSUE THREAD

**Issue by deemeetree** [cytoscape/cytoscape.js#2345]
Possible incorrect betweenness centrality measure

Bug report

**Environment info**

- Cytoscape.js version : 3.5.0
- Browser/Node.js & version : Node.Js 8.8.0

**Current (buggy) behaviour**

Currently [betweennessCentrality](http://js.cytoscape.org/#eles.betweennessCentrality) for directed/weighted graphs is calculated in such a way that if the graph is directed and the weight function is calculated then the nodes with the less weight are given a higher betweenness centrality score, which is not logical. 

**Desired behaviour**

Nodes that have edges with higher weight should have higher betweenness centrality measure than a similarly embedded nodes with a lower weight.

**Comment by maxkfranz** [comment]
Are your weights positive?  Can you provide a sample dataset?

**Comment by deemeetree** [comment]
Yes, they are. Here is my dataset: 
https://www.dropbox.com/s/zsmq57ofv0d8w7e/cytoscape-infranodus-elements.json?dl=0

I call this object `elesnodes`

Then I create an object using 
```var eles = cy.add(elesnodes);```

And here's my Javascript code to calculate betweenness centrality:

```
     var bc = eles.bc({
          weight: function(edges) {
              return edges.data('weight');
          },
          directed: true
      });
```
and I calculate betweenness centrality for each using 

```
          var nn = eles.getElementById(cy.nodes()[i].data('id'));

          var currentbc = bc.betweennessNormalized(nn);
```

**Comment by deemeetree** [comment]
You might be interested in this discussion on the same subject we're having on `networkx` python library. Both cytoscape and networkx seem to function based on the assumption that higher weight is more expensive, which is not always the case in biology (think of synapses). 

https://github.com/networkx/networkx/issues/3369

**Comment by deemeetree** [comment]
@maxkfranz did you have a chance to check this issue?

**Comment by maxkfranz** [comment]
It seems like it's just an issue of the ambiguity of the word "weight".  The word "weight" just means a number.  It doesn't necessarily mean that a higher weight is better or stronger; that depends on the context.

Some things like edge-weighted force-directed layouts will have the semantics of a larger weight meaning a stronger value.  Some other things like path-finding algorithms use weight as a cost value.

These contexts could be made clearer in the documentation.

Basically if your concept of weight doesn't match up in the right direction for the particular context, then you need to invert the weights somehow.  That's something you probably have to adjust yourself for the particular algorithm.  I don't think it's generally automatable.
